logging/telemetryConfig.go

```diff
@@ -120,6 +130,22 @@ func (cfg TelemetryConfig) getInstanceName() string {
 	return fmt.Sprintf("%s:", pathHashStr[:16])
 }
 
+func SanitizeTelemetryString(input string, maxParts int) string {
+	parts := strings.SplitN(input, ":", maxParts)
+	for i := range parts {
+		// Limit to alphanumeric, and _-
+		if reg != nil {
```

Comment:
nit: move the `reg != nil` outside the for loop. In practice, it will never happen, but we have to do the check.

Reply:
I'm not sure it's worth it, or what the error handling would even be. The current logic just skips the regex if it doesn't exist, but also supports removing colons from last part of the string split, so `guid-pattern:removes-the-:colon` -> `guid-pattern:removes-the-colon`.

Reply:
Actually, we definitely can't remove colons, the IPv6 address format will have lots of them!
I'll apply the regex replace all outside the loop.